src/lib/main/core/body/td-core.directive.ts

```diff
@@ -41,7 +45,18 @@ export class TdCoreDirective implements Td, OnInit, OnDestroy {
 		link(this.viewContainerRef, this);
 	}
 
-	mode(value: 'view' | 'edit') {
+	ngOnChanges(changes: SimpleChanges) {
+		for (const propName of Object.keys(changes)) {
```

Comment:
just get necessary changes, don't go through the collection, it's performance critical